docs/ldml/tr35-keyboards.md

```diff
@@ -2412,6 +2416,287 @@ Used in the `to=`
 
     Emits the named mark. Also see [Markers](#markers).
 
+#### Transform Grammar
+
+##### Transform From Grammar
+
+The `from=` attribute MUST match the `from-match` rule in this grammar. Not all strings which match this grammar are valid:
+
+* It is an error if there are more than 9 capture groups
+* Unicode codepoints and escaped characters may not be invalid or unpaired surrogates
+* The CLDR repository may define additional requirements on the repertoire, such as requiring all characters to be in a published Unicode version and disallowing private-use characters.
+* It is an error to reference variables that are not defined.
+
+The following is the [LDML EBNF](./tr35.md#ebnf) format for the grammar:
+
+```ebnf
+
+from-match
+         ::= '^'? atoms
+atoms    ::= atom ( '|'? atom )*
+atom     ::= quark quantifier?
+quark    ::= non-group
+           | group
+non-group
+         ::= simple-matcher
+           | codepointseq
+           | variable
+variable ::= string-variable
+           | set-variable
+string-variable
+         ::= '${' var-id '}'
+set-variable
+         ::= '$[' var-id ']'
+group    ::= capturing-group
+           | non-capturing-group
+quantifier
+         ::= bounded-quantifier
+           | '?'
+codepointseq
```

Comment:
how is `codepointseq` different from `codepoint`?

Reply:
The spec allows for a space separated list of code points… However, where specifying a character class range start or end, only a single code point must be allowed, that's why there are two different rules. 

I'll clarify this.